base/src/com/google/idea/blaze/base/lang/buildfile/actions/BuildFileModifierImpl.java

```diff
@@ -53,7 +67,110 @@ public boolean addRule(Project project, Label newRule, Kind ruleKind) {
     return true;
   }
 
-  private PsiElement createRule(Project project, Kind ruleKind, String ruleName) {
+  @Override
+  public boolean addDepToRule(Project project,
+                              Label targetToBeAdded,
+                              VirtualFile virtualFileToBeEdited) {
+    boolean wasDepsAddedToRule = false;
+
+    Optional<BlazePackage> parentPackage = Optional.of(BuildFileUtils.getBuildFile(project, virtualFileToBeEdited));
+    if (parentPackage.isPresent()) {
+        Optional<PsiElement> targetToBeEdited =
+              Optional.fromNullable(
+                      BuildFileUtils.findBuildTarget(
+                              project,
+                              parentPackage.get(),
+                              new File(virtualFileToBeEdited.getPath())
+                      )
+              );
+        if(targetToBeEdited.isPresent()){
+            java.util.Optional<Argument.Keyword> maybeDepsKeyword = findTargetDepsAttribute(targetToBeEdited.get());
+            if(!maybeDepsKeyword.isPresent()){
+                maybeDepsKeyword = addDepsAttribute(project, targetToBeEdited.get());
+            }
+            addTargetToDepsIfNeeded(maybeDepsKeyword.get(), targetToBeAdded, project);
+            wasDepsAddedToRule = true;
+        }
+    }
+
+    return wasDepsAddedToRule;
+  }
+
+    @NotNull
+    private java.util.Optional<Argument.Keyword> addDepsAttribute(Project project, PsiElement targetToBeEdited) {
+        java.util.Optional<Argument.Keyword> maybeDepsKeyword;
+        ArgumentList targetAttributes = PsiUtils.findFirstChildOfClassRecursive(targetToBeEdited, ArgumentList.class);
+        addDepsAttributeToTargetAttributes(targetAttributes, project);
+        maybeDepsKeyword = findTargetDepsAttribute(targetToBeEdited);
+        PsiElement whiteSpaceLiteral = PsiParserFacade.SERVICE.getInstance(project).createWhiteSpaceFromText("\n    ");
+        targetAttributes.addBefore(whiteSpaceLiteral, maybeDepsKeyword.get());
+        PsiElement commaLiteral = createCommaElement(project);
+        targetAttributes.addAfter(commaLiteral, maybeDepsKeyword.get());
+
+        return maybeDepsKeyword;
+    }
+
+    @NotNull
+    private java.util.Optional<Argument.Keyword> findTargetDepsAttribute(PsiElement psiElement) {
+        List<Argument.Keyword> allChildKeywords = PsiUtils.findAllChildrenOfClassRecursive(psiElement, Argument.Keyword.class);
+        Stream<Argument.Keyword> depsKeywords = allChildKeywords.stream().filter(keyword -> keyword.getName().equals("deps"));
+        return depsKeywords.findFirst();
+    }
+
+    private void addTargetToDepsIfNeeded(Argument.Keyword depsKeyword, Label targetToBeAdded, Project project) {
+        PsiElement depsElement = depsKeyword.getPsiChild(BuildElementTypes.LIST_LITERAL, null);
+        if(depsDoesntHaveWhiteSpace(depsElement)){
+            addWhiteSpaceElement(project, depsElement, "\n    ");
```

Comment:
Can we use https://github.com/bazelbuild/intellij/blob/master/base/src/com/google/idea/blaze/base/buildmodifier/BuildFileFormatter.java for final formatting?

Reply:
todo